tracer/test/Datadog.Trace.ClrProfiler.IntegrationTests/CI/PipesXUnitTests.cs

```diff
@@ -0,0 +1,288 @@
+// <copyright file="PipesXUnitTests.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Datadog.Trace.Ci;
+using Datadog.Trace.Ci.Tags;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.ExtensionMethods;
+using Datadog.Trace.TestHelpers;
+using Datadog.Trace.Vendors.Newtonsoft.Json;
+using VerifyXunit;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Datadog.Trace.ClrProfiler.IntegrationTests.CI
+{
+    [UsesVerify]
+    public class PipesXUnitTests : TestingFrameworkTest
+    {
+        private const string TestBundleName = "Samples.XUnitTests";
+        private const string TestSuiteName = "Samples.XUnitTests.TestSuite";
+        private const string UnSkippableSuiteName = "Samples.XUnitTests.UnSkippableSuite";
+        private const int ExpectedSpanCount = 16;
+
+        public PipesXUnitTests(ITestOutputHelper output)
+            : base("XUnitTests", output)
+        {
+            SetServiceName("xunit-tests");
+            SetServiceVersion("1.0.0");
+        }
+
+        [SkippableTheory]
+        [MemberData(nameof(PackageVersions.XUnit), MemberType = typeof(PackageVersions))]
+        [Trait("Category", "EndToEnd")]
+        [Trait("Category", "TestIntegrations")]
+        public async Task SubmitTraces(string packageVersion)
```

Comment:
If these tests are identical as existing apart from transport, can't we extract that to a base class, 300 lines of copied code seems a bit excessive 😄 

Also, I can practically guarantee this will flake on named pipes - I suggest you wrap it in our usual auto-retries for pipes

Reply:
True, done in https://github.com/DataDog/dd-trace-dotnet/pull/5634/commits/5f70ddf7aac47e740c4b9a0fe61ae5728af8a310